graphql/schema/mutation.graphql

```diff
@@ -42,10 +42,10 @@ type Mutation {
   saveDistro(opts: SaveDistroInput!): SaveDistroPayload! # Has directive on SaveDistroInput.
 
   # hosts
-  reprovisionToNew(hostIds: [String!]!): Int!
-  restartJasper(hostIds: [String!]!): Int!
+  reprovisionToNew(hostIds: [String!]! @requireHostAccess(access: EDIT)): Int!
+  restartJasper(hostIds: [String!]! @requireHostAccess(access: EDIT)): Int!
   updateHostStatus(
-    hostIds: [String!]!
+    hostIds: [String!]! @requireHostAccess(access: EDIT)
```

Comment:
I think you're able to click these buttons on your own spawn hosts too so they might be `SPAWN_EDIT` (but you likely wouldn't have to worry about that with the previous comment)